shark/src/main/java/shark/ObfuscationChecker.kt

```diff
@@ -0,0 +1,22 @@
+package shark
+
+interface ObfuscationChecker {
+  fun isCodeObfuscated(): Boolean
+}
+
+class ReflectionObfuscationChecker : ObfuscationChecker {
+
+  /**
+   * This function uses reflection to find field by name in a given class.
+   * If code is obfuscated then this lookup will fail.
+   */
+  override fun isCodeObfuscated(): Boolean {
+    return try {
+      Class.forName("shark.LeakTrace")
+          .getDeclaredField("gcRootType")
```

Comment:
I see two risks here: 1) we're checking against a public API and sometimes people put rules excluding libs like LeakCanary. 2) We're creating hidden coupling between LeakTrace and ReflectionObfuscationChecker. What if the class / field name change?

Should we create a dedicated dumb (internal) class in a completely different package (ie not shark), for the purpose of checking for proguard? That way the class can be commented and it's purpose is clear.